src/plots/plots.js

```diff
@@ -1955,6 +1955,21 @@ plots.autoMargin = function(gd, id, o) {
     }
 };
 
+plots.redrawWithShift = function(gd) {
```

Comment:
This looks like a fine workaround for now. But if you're only going to use this function inside the file where it's defined, don't export it - makes refactoring easier, and minifies better. Also as named it sounds like a verb, like this function is doing the redraw. I'd name it something like `needsRedrawForShift`.